src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -198,9 +198,20 @@ def convert_to_variant(filename):
             variant_filename = f"{filename.split('.')[0]}.{variant}.{filename.split('.')[1]}"
         return variant_filename
 
+    components_with_variant = set()
+    for filename in variant_filenames:
+        if not len(filename.split("/")) == 2:
+            continue
+        component, component_filename = filename.split("/")
+        components_with_variant.add(component)
```

Comment:
```suggestion
        components_with_variant.add(filename.split("/")[0])
```

I think we should match the logic to find "component" between variant_filenames and non_variant_filenames

Reply:
ohh I realize you actually just updated this to account for files that are not put inside any subfolders 
maybe make this a in-line function `find_component` or something and do the same the for f in non_variant_filenames too